pkg/cloudprovider/vsphere/route/routeprovider.go

```diff
@@ -76,7 +76,11 @@ func (p *routeProvider) ListRoutes(ctx context.Context, clusterName string) ([]*
 		klog.Errorf("querying static routes for cluster %s failed", clusterName)
 		return nil, err
 	}
+	return p.generateRoutes(staticRoutes), nil
```

Comment:
if the SearchResponse for querying staticRoutes is empty, maybe we should skip going through `generateRoutes` function. 
wdyt?